app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -373,28 +377,29 @@ public function uploadShapefile(Request $request)
 
                     return response()->json(['error' => 'Failed to convert Shapefile to GeoJSON', 'message' => $process->getErrorOutput()], 500);
                 }
+                $geojsonContent = file_get_contents($geojsonPath);
+                $redis_key = 'shapefile_file_' . uniqid();
+                Redis::set($redis_key, $geojsonContent, 'EX', 7200);
+                $delayedJob = DelayedJob::create(['status' => InsertGeojsonToDBJob::STATUS_PENDING]);
 
                 $polygonLoadedList = isset($body['polygon_loaded']) && filter_var($body['polygon_loaded'], FILTER_VALIDATE_BOOLEAN);
                 $submitPolygonsLoaded = $request->input('submit_polygon_loaded') === true || $request->input('submit_polygon_loaded') === 'true';
-                if (! $polygonLoadedList && ! $submitPolygonsLoaded) {
-                    $job = new InsertGeojsonToDBJob($geojsonFilename, $site_id, 'site', $body['primary_uuid'] ?? null);
-                }
+
+                if (! $polygonLoadedList && $submitPolygonsLoaded) {
+                  $job = new InsertGeojsonToDBJob($redis_key, $delayedJob->id, $site_id, 'site', $body['primary_uuid'] ?? null, $body['submit_polygon_loaded']);
+                } 
 
                 if ($polygonLoadedList) {
-                    $filePath = $tempDir . DIRECTORY_SEPARATOR . $geojsonFilename;
-                    $geojsonContent = file_get_contents($filePath);
                     $polygonLoaded = $this->GetAllPolygonsLoaded($geojsonContent, $site_id);
-
                     return response()->json($polygonLoaded->original, 200);
                 }
 
                 if ($submitPolygonsLoaded) {
-                    $job = new InsertGeojsonToDBJob($geojsonFilename, $site_id, 'site', $body['primary_uuid'] ?? null, $body['submit_polygon_loaded']);
+                    $job = new InsertGeojsonToDBJob($redis_key, $delayedJob->id,  $site_id, 'site', $body['primary_uuid'] ?? null, $body['submit_polygon_loaded']);
```

Comment:
Looks like this method has the same structure problems as the one above.

Reply:
done